tools/system_calls.py

```diff
@@ -161,3 +162,43 @@ def get_screen():
     buffered_screen = BytesIO()
     screen.save(buffered_screen, format='JPEG')
     return base64.b64encode(buffered_screen.getvalue()).decode('utf-8')
+
+
+def url_history(url):
+    """ Saves casted url in file
+
+    """
+
+    with open('/var/lib/teleserver/app/config_teleserver.yml', 'r') as file:
+        urls = yaml.load(file, Loader=yaml.FullLoader)
+        urls_to_hist = urls.get("urls")
+        urls_config = urls.get("url_config")
+
+    if urls_to_hist is not None:
+        if len(urls_to_hist) < urls_config:
+            urls_to_hist.append(urls_to_hist[len(urls_to_hist) - 1])
+            for x in range(len(urls_to_hist)-1, -1, -1):
+                urls_to_hist[x] = urls_to_hist[x-1]
+        else:
+            if len(urls_to_hist) > urls_config:
+                del urls_to_hist[urls_config:len(urls_to_hist)]
+            for x in range((urls_config-1), -1, -1):
+                urls_to_hist[x] = urls_to_hist[x-1]
+        urls_to_hist[0] = url
+    else:
+        urls_to_hist = [url]
```

Comment:
This whole block can be replaced with simple queue:
```
from collections import deque
```
and assuming that you have variable with max length defined and data loaded as list
```
historic_urls = deque(<variable with loaded data>, maxlen=<variable of max length>)
historic_urls.append(<variable with url received from input>)
```

That way you don't need to implement your own mechanism of handling classic queue and it's more elegant and safe.
Also deque handles by itself or problems with any additonal urls that should be deleted or length of urls.